workers/gossip-server/src/peer_discovery/heartbeat.rs

```diff
@@ -123,18 +129,30 @@ impl GossipProtocolExecutor {
         let cluster_id = self.global_state.get_cluster_id().await?;
         let same_cluster = peer_info.get_cluster_id() == cluster_id;
 
-        let now = get_current_time_seconds();
+        let now = get_current_time_millis();
         let last_heartbeat = now - peer_info.get_last_heartbeat();
 
         #[allow(clippy::if_same_then_else)]
-        if same_cluster && last_heartbeat < CLUSTER_HEARTBEAT_FAILURE_MS / 1000 {
+        if same_cluster && last_heartbeat < CLUSTER_HEARTBEAT_FAILURE_MS {
             return Ok(());
-        } else if !same_cluster && last_heartbeat < HEARTBEAT_FAILURE_MS / 1000 {
+        } else if !same_cluster && last_heartbeat < HEARTBEAT_FAILURE_MS {
             return Ok(());
         }
 
+        // TODO: Give the cluster time to reject peer expiry
+
+        // Send a peer expiry proposal to the cluster
```

Comment:
per our convo internally, this pubsub message should only be sent if `same_cluster == true`, right?